### ISSUE TITLE
Does not build with Intel Fortran

### ISSUE THREAD

**Issue by dbischof90** [pdfo/pdfo#69]
Hi! I was curious about your improvements on LINCOA and put it into a separate Visual Studio Fortran project with the Intel compiler.

I'm getting the following issue:
```
Compiling with Intel® Fortran Compiler Classic 2021.10.0 [Intel(R) 64]...
[...]
C:\[...]\lincoa\lincoa.f(199): error #6633: The type of the actual argument differs from the type of the dummy argument.   [W]
     2  W(IPQ),W(IBMAT),W(IZMAT),NDIM,W(ISTP),W(ISP),W(IXN),W(IAC),
------------------------------------------------------------^
compilation aborted for C:\[...]\lincoa\lincoa.f (code 1)
```
Background: I'm currently trying to call your routines from C#.
Any ideas?

**Comment by ragonneau** [comment]
Hi @dbischof90,

Sorry for not getting back to you sooner. Could you provide more details on your compilation (e.g., the complete command, the whole output, etc)? I am not sure what is the problem using only the info here...

Meanwhile, if you are interested in Powell methods, you may have a look at https://github.com/libprima/prima. It is a modernization of the 5 DFO methods of Powell.

Look forward for your reply,
Cheers,
Tom.

**Comment by dbischof90** [comment]
Hi @ragonneau!
No worries, I did indeed also discuss this with the maintainters in https://github.com/libprima/prima/issues/103 and I made that one compiling (in the sense that I get a DLL) and when running the examples I am also able to get that going but I'm unable to embedd this into my C# project - my best guess is that icl/ifort and msvc do not play ball together, the DLL is built with the former and the rest of the project uses the latter.

I fell back to using an inferior version which was rewritten in C# with the same limitations as the original FORTRAN77 version and will keep an eye on the project if someone manages to write an intuitive DotNET wrapper. Until then I need to be aware that the implementations are a bit faulty in known corner cases.
